### DISCUSSION TITLE
Adguard home all interface

### DISCUSSION THREAD

**Discussion by Hanif-Faez** [trinib/Adguard-Wireguard-Unbound-Cloudflare#58] [Help]
### Operating System

Raspberry Pi

### Architecture

64-bit

### Platform

Linux

### Project

Aduard Home

### Browser

Chrome

### Issue

Not working

### Issue Description

Hi.

Currently I'm using pi zero 2w.
Is it possible to let Adguard listen on all interface since tailscale not working when listening to only wlan0. Just like pihole permit all origin? 
I already tried editing adguard.yaml and set dns interface to 0.0.0.0 and dns service won't start.

**Comment by welcome** [comment]
_<h3>Thanks for opening your first issue here 🙋🕵️</h3>_
<img src="https://media.giphy.com/media/10UeedrT5MIfPG/giphy.gif" width="40%">

**Comment by trinib** [comment]
I see.. yes try all interface

does that settings works for you?

**Comment by trinib** [comment]
I think it might need to listen to local lan. do you get a bind host error?

**Comment by Hanif-Faez** [comment]
> I see.. yes try all interface
> 
> does that settings works for you?

I already tried listening to all interfaces with 0.0.0.0 but after reloading Adguard home, DNS service not working it spinning continuously.
It only work when I followed your instruction with DNS interface wlan0 since my pi is wireless.

**Comment by trinib** [comment]
> > I see.. yes try all interface
> > does that settings works for you?
> 
> I already tried listening to all interfaces with 0.0.0.0 but after reloading Adguard home, DNS service not working it spinning continuously. It only work when I followed your instruction with DNS interface wlan0 since my pi is wireless.

hmm i never tried with wifi..try making ip static on pi .

**Comment by trinib** [comment]
@Haniffaez your issue is tailscale not working with AGH using wlan0 right ?.. I wonder if something else is listening on address

**Comment by Hanif-Faez** [comment]
> @Haniffaez your issue is tailscale not working with AGH using wlan0 right ?.. I wonder if something else is listening on address

Ya tailscale can't use DNS from Adguard if I set to wlan0. It need to listen on all interface just like pihole permit all origin. My pi IP already reserved on router. I follow this step on tailscale website.
https://tailscale.com/kb/1114/pi-hole/

**Comment by trinib** [comment]
did you check to see if anything else on port 53?

```
sudo netstat -anp | grep 53
```

**Comment by trinib** [comment]
checking adguard logs as well can be helpful

**Comment by Hanif-Faez** [comment]
Sorry for late reply. 
Here is the output.
![Screenshots_2022-06-13-09-34-09](https://user-images.githubusercontent.com/98368603/173264581-d28c557c-de12-478b-93e5-b20515f3c990.png)
What am I looking at, I'm new to this.
Do I need to enable verbosity and try again?

**Comment by trinib** [comment]
what you get for 

`sudo systemctl status AdGuardHome`

&

`sudo nano /var/log/AdGuardHome.err`

**Comment by Hanif-Faez** [comment]
> what you get for
> 
> `sudo systemctl status AdGuardHome`
> 
> &
> 

> `sudo nano /var/log/AdGuardHome.err`

Here is the output after I stopped Adguard service and went to Adguard.yaml to change the DNS interface to 0.0.0.0
![1](https://user-images.githubusercontent.com/98368603/173274577-458bdc30-7227-484a-84f1-46c2f2836ad6.png)

Second output: sudo systemctl status AdGuardHome
![error](https://user-images.githubusercontent.com/98368603/173274745-0744becf-d416-4209-ab86-a3d7e67e718c.png)

DNS service won't starts if it is set to 0.0.0.0 and the whole network is down
![error 3](https://user-images.githubusercontent.com/98368603/173274982-f8a4b3d1-fd84-4f43-883a-02af5144eb62.png)

Here is the Adguard.yaml
![error 2](https://user-images.githubusercontent.com/98368603/173275080-8894837c-2076-43bb-8044-d4b060e6fe36.png)

**Comment by trinib** [comment]
hmmm try setting up firewall rules https://help.ubuntu.com/community/UFW
```
sudo apt install ufw -y
```
```
sudo ufw enable
```
```
sudo ufw allow from 0.0.0.0
```
```
sudo ufw allow in on wlan0
```

**Comment by Hanif-Faez** [comment]
> 

Sorry I have to reinstall the whole pi after messing up with ufw.
After reinstalling and configuring ufw, DNS service still can't start when set to 0.0.0.0
When checking for Adguard status, the service run but it keeps loading just like the screenshot I attached before.

![err](https://user-images.githubusercontent.com/98368603/173535241-e4224012-adda-4a1f-909b-d6beca1da00a.png)

**Comment by trinib** [comment]
@Haniffaez have you tried with a wired connection to see if it works?  then you can know if its a wifi issue or not

**Comment by trinib** [comment]
when i use all interface i get in bind host 0.0.0.0

![image](https://user-images.githubusercontent.com/18756975/173577257-f8a7e3c2-ac06-4415-b714-7995ab51661e.png)


did you change that ?

![image](https://user-images.githubusercontent.com/18756975/173577457-4a5a2ea0-8186-4da9-8e3c-0237db4ade2d.png)

**Comment by Hanif-Faez** [comment]
> @Haniffaez have you tried with a wired connection to see if it works? then you can know if its a wifi issue or not

Ya wired connection might be the last resort but I need to get ethernet to micro usb adapter and see how it goes.

**Comment by Hanif-Faez** [comment]
> when i use all interface i get in bind host 0.0.0.0
> 
> ![image](https://user-images.githubusercontent.com/18756975/173577257-f8a7e3c2-ac06-4415-b714-7995ab51661e.png)
> 
> did you change that ?
> 
> ![image](https://user-images.githubusercontent.com/18756975/173577457-4a5a2ea0-8186-4da9-8e3c-0237db4ade2d.png)

Ya it does not work when I put 0.0.0.0 and DNS service seems like not even start up by looking at continuous loading icon on adguard dashboard. All interface only play nice when I use default upstream dns like cloudflare or Google dns without unbound and TLS forwarder.

**Comment by trinib** [comment]
so that 192.168.1.11 address is for tailscale to work ?

**Comment by Hanif-Faez** [comment]
> so that 192.168.1.11 address is for tailscale to work ?

I used default upstream dns (cloudflare) before and it work well with tailscale but it need to listen on all interface. When I use unbound and TLS forwarder, set interface to 0.0.0.0 the dns service won't start and it keep loading until I revert back changes to 192.168.1.11

**Comment by Hanif-Faez** [comment]
> > > when i use all interface i get in bind host 0.0.0.0
> > > ![image](https://user-images.githubusercontent.com/18756975/173577257-f8a7e3c2-ac06-4415-b714-7995ab51661e.png)
> > > did you change that ?
> > > ![image](https://user-images.githubusercontent.com/18756975/173577457-4a5a2ea0-8186-4da9-8e3c-0237db4ade2d.png)
> > 
> > 
> > Ya it does not work when I put 0.0.0.0 and DNS service seems like not even start up by looking at continuous loading icon on adguard dashboard. All interface only play nice when I use default upstream dns like cloudflare or Google dns without unbound and TLS forwarder.
> 
> sudo nano /etc/resolv.conf nameserver 127.0.0.1 save and exit sudo systemctl restart systemd-resolved
> 
> now you can use bind host 0.0.0.0 for AdGuardHome
> 
> sudo systemctl status systemd-resolved ● systemd-resolved.service - Network Name Resolution Loaded: loaded (/lib/systemd/system/systemd-resolved.service; enabled; vendor preset: enabled) Active: active (running) since Tue 2022-06-14 19:19:15 EEST; 2h 30min ago Docs: man:systemd-resolved.service(8) man:org.freedesktop.resolve1(5) https://www.freedesktop.org/wiki/Software/systemd/writing-network-configuration-managers https://www.freedesktop.org/wiki/Software/systemd/writing-resolver-clients Main PID: 746 (systemd-resolve) Status: "Processing requests..." Tasks: 1 (limit: 4379) Memory: 7.8M CPU: 146ms CGroup: /system.slice/systemd-resolved.service └─746 /lib/systemd/systemd-resolved
> 
> Jun 14 19:19:15 homeserversystemd[1]: Starting Network Name Resolution... Jun 14 19:19:15 homeserver systemd-resolved[746]: Positive Trust Anchors: Jun 14 19:19:15 homeserver systemd-resolved[746]: Negative trust anchors: home.arpa 10.in-addr.arpa 16.172.in-addr.arpa 17.172.in-addr.arpa 18.1> Jun 14 19:19:15homeserver systemd-resolved[746]: Using system hostname '-homeserver'. Jun 14 19:19:15 homeserver systemd[1]: Started Network Name Resolution.
> 
> ![2022-06-14_21-49-00](https://user-images.githubusercontent.com/64448768/173666836-37f17bda-6a3b-434a-8c6a-aa2adcb5ae45.jpg) ![2022-06-14_21-51-39](https://user-images.githubusercontent.com/64448768/173666842-dbeffef2-7e3b-47bc-ac1d-6d221c3f8910.jpg)

I guess mine already set to 127.0.0.1

![Screenshots_2022-06-15-15-28-46](https://user-images.githubusercontent.com/98368603/173768855-7cff5349-88fa-4924-9bb9-06f834caaf3f.png)

**Comment by trinib** [comment]
> > so that 192.168.1.11 address is for tailscale to work ?
> 
> I used default upstream dns (cloudflare) before and it work well with tailscale but it need to listen on all interface. When I use unbound and TLS forwarder, set interface to 0.0.0.0 the dns service won't start and it keep loading until I revert back changes to 192.168.1.11

its a long shot but try adding `nameserver 192.168.1.11` in resolv.conf

```
sudo apt install resolvconf
```
```
sudo systemctl start resolvconf.service
sudo systemctl enable resolvconf.service
sudo systemctl status resolvconf.service
```
```
sudo nano /etc/resolvconf/resolv.conf.d/head
```
add `nameserver 192.168.1.11`

```
sudo systemctl restart resolvconf.service
sudo systemctl restart systemd-resolved.service
```

check `sudo nano /etc/resolv.conf` 

restart pi just in case

see if any change with unbound etc🙏

if not you will have to ask on https://github.com/AdguardTeam/AdGuardHome

**Comment by trinib** [comment]
@Haniffaez or try `nameserver 0.0.0.0`  if nameserver 192.168.1.11 do not work

**Comment by Hanif-Faez** [comment]
> > > so that 192.168.1.11 address is for tailscale to work ?
> > 
> > 
> > I used default upstream dns (cloudflare) before and it work well with tailscale but it need to listen on all interface. When I use unbound and TLS forwarder, set interface to 0.0.0.0 the dns service won't start and it keep loading until I revert back changes to 192.168.1.11
> 
> its a long shot but try adding `nameserver 192.168.1.11` in resolv.conf
> 
> ```
> sudo apt install resolvconf
> ```
> 
> ```
> sudo systemctl start resolvconf.service
> sudo systemctl enable resolvconf.service
> sudo systemctl status resolvconf.service
> ```
> 
> ```
> sudo nano /etc/resolvconf/resolv.conf.d/head
> ```
> 
> add `nameserver 192.168.1.11`
> 
> ```
> sudo systemctl restart resolvconf.service
> sudo systemctl restart systemd-resolved.service
> ```
> 
> check `sudo nano /etc/resolv.conf`
> 
> restart pi just in case
> 
> see if any change with unbound etc🙏
> 
> if not you will have to ask on https://github.com/AdguardTeam/AdGuardHome

I have tried all the steps but it still not working after adding nameserver and editing Adguard.yaml to listen on 0.0.0.0
Got the same error DNS server is starting up and it keeps loading.

![3](https://user-images.githubusercontent.com/98368603/173782551-9e946ec3-b549-44a5-8fa2-63db0767a4a2.png)

Never mind, I just use this without tailscale or went back to default upstream DNS if I need.

**Comment by Hanif-Faez** [comment]
Wait can I close this? Sorry, this is my first post. I guess need to refer Adguardhome.

**Comment by trinib** [comment]
@Haniffaez if you found solution from AGH community.. can you please mention solution here and then you can close .. thanks

**Comment by ryanwalexander** [comment]
I have the same issue trying to use adguard home.  Switched to pi-hole and it works fine using it.
